core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -41,8 +41,20 @@ private[spark] class SparkDeploySchedulerBackend(
 
   private val registrationBarrier = new Semaphore(0)
 
-  private val maxCores = conf.getOption("spark.cores.max").map(_.toInt)
-  private val totalExpectedCores = maxCores.getOrElse(0)
+  val maxCores = conf.getOption("spark.cores.max").map(_.toInt)
+  val coreNumPerTask = conf.getInt("spark.task.cpus", 1)
+    
+  if (coreNumPerTask < 1) {
+    throw new IllegalArgumentException(
+      s"spark.task.cpus is set to an invalid value $coreNumPerTask")
+  }
```

Comment:
The right place to do this is in `TaskSchedulerImpl`. Also we should be more specific in the error message what "invalid value" means:
```
"spark.task.cpus" must be greater than 0! (was $coresPerTask)
```